source/agora/consensus/protocol/Nominator.d

```diff
@@ -556,6 +563,61 @@ extern(D):
         this.active_timers[type] = this.taskman.setTimer(
             timeout.msecs, { callCPPDelegate(callback); });
     }
+
+    /***************************************************************************
+
+        Used by the nomination protocol to randomize the order of messages
+        between nodes.
+
+        Params:
+            slot_idx = the slot index we're currently reaching consensus for.
+            prev = the previous data set for the provided slot index.
+            is_priority = the flag to check that this call is for priority.
+            round_number = the nomination round
+            node_id = the id of the node for which this computation is being made
+
+    ***************************************************************************/
+
+    public override uint64_t computeHashNode(uint64_t slot_idx,
+        ref const(Value) prev, bool is_priority, int32_t round_number,
+        ref const(NodeID) node_id) nothrow
+    {
+        uint512 hash;
+        const random_seed = this.ledger.getValidatorRandomSeed(Height(slot_idx));
+        try
+        {
+            hash = getHashOf(slot_idx, prev, is_priority ? hash_P : hash_N,
+                round_number, node_id, random_seed);
+        }
+        catch (Exception ex)
+        {
+            log.fatal("Computing hash of the node({}) failed: {}, Data was: ",
+                node_id, ex.msg);
+            log.fatal("\tslot_idx: {}, prev: {}, is_priority: {}, randon_seed: {}",
```

Comment:
```suggestion
            log.fatal("\tslot_idx: {}, prev: {}, is_priority: {}, random_seed: {}",
```

Why isn't this all in one log call? You can add `\n` if you really want to add a newline.